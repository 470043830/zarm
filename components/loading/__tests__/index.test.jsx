import React from 'react';
import { render, mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from '../index';

describe('Confirm', () => {
  const props = {
    mask: true,
    stayTime: 1,
    onMaskClick: jest.fn(),
  };
  it('renders correctly', () => {
    const wrapper = render(
      <Loading {...props}>foo</Loading>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('visible change false', () => {
    const wrapper = mount(
      <Loading visible>foo</Loading>,
    );
    wrapper.setProps({ visible: false });
  });

  it('stayTime', () => {
    jest.useFakeTimers();
    props.onClose = jest.fn();
    const wrapper = mount(
      <Loading stayTime={5000} visible {...props}>foo</Loading>,
    );
    jest.runAllTimers();
    wrapper.unmount();
  });

  it('stayTime is 0', () => {
    const wrapper = shallow(
      <Loading stayTime={0}>foo</Loading>,
    );
    wrapper.setProps({ visible: true });
  });

  it('afterClose', () => {
    jest.useFakeTimers();
    const afterClose = jest.fn();
    const wrapper = mount(
      <Loading stayTime={5000} visible {...props} afterClose={afterClose}>foo</Loading>,
    );
    jest.runAllTimers();
    // expect(afterClose).toHaveBeenCalled();
    wrapper.unmount();
  });

  it('static function show', () => {
    const wrapper = Loading.show();
    expect(wrapper).toMatchSnapshot();
  });

  it('static function hide', () => {
    const wrapper = Loading.hide();
    expect(wrapper).toMatchSnapshot();
  });
});
