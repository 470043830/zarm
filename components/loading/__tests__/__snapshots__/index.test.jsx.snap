// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm renders correctly 1`] = `
<div
  class="za-loading za-loading--open"
>
  <div
    class="za-loading__container"
  >
    <span
      class="za-activity-indicator za-activity-indicator--lg za-activity-indicator__circular--loading"
    >
      <svg
        class="za-activity-indicator__circular"
        viewBox="31, 31 62 62"
      >
        <circle
          cx="62"
          cy="62"
          fill="none"
          r="28.5"
          style="stroke-width:5"
        />
      </svg>
    </span>
  </div>
</div>
`;
